tools/expire/main.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"context"
+	"flag"
+
+	"github.com/distributedio/titan/db"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/store/tikv"
+	"go.uber.org/zap"
+)
+
+var log = zap.L()
+
+type option struct {
+	db        int
+	namespace string
+	batch     int
+}
+
+func cleanUp(txn kv.Transaction, mkey kv.Key, database *db.DB,
+	obj *db.Object) error {
+	switch obj.Type {
+	case db.ObjectString:
+		return txn.Delete(mkey)
+	case db.ObjectZSet:
+		dkey := db.DataKey(database, obj.ID)
+		skey := db.ZSetScorePrefix(database, obj.ID)
+		txn.Delete(dkey) // ignore its error for simplicity
+		return txn.Delete(skey)
+	default:
+		dkey := db.DataKey(database, obj.ID)
+		return txn.Delete(dkey)
+	}
+	return nil
+}
+
+func doExpire(s kv.Storage, database *db.DB, start kv.Key, limit int) (kv.Key, error) {
+	txn, err := s.Begin()
+	if err != nil {
+		return nil, err
+	}
+	iter, err := txn.Iter(start, nil)
```

Comment:
defined end prefix key
 kv.Key(start).PrefixNext()

Reply:
I want to scan the whole databases, so I did not want a prefix-next.